src/ui/hooks/recordings.ts

```diff
@@ -734,3 +734,48 @@ export function useUpdateRecordingTitle() {
 
   return updateRecordingTitle;
 }
+
+export async function getRecordingMetadata(id: string) {
+  const resp = await fetch(process.env.NEXT_PUBLIC_API_URL!, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      query: `
+        query GetRecording($recordingId: UUID!) {
+          recording(uuid: $recordingId) {
+            uuid
+            url
+            title
+            duration
+            private
+            isInitialized
+            owner {
+              name
+            }
+            workspace {
+              name
+            }
+          }
+        }
+      `,
+      variables: {
+        recordingId: id,
+      },
+    }),
+  });
+
+  const json: {
+    data: { recording: Recording & { owner?: { name: string } } };
+    error: any;
+  } = await resp.json();
```

Comment:
I think we need to gracefully handle errors here. Right now the preview branch crashes on private recordings: https://devtools-ewr79cidf-recordreplay.vercel.app/recording/823241b9-54fd-4cc4-b315-3937694c0b0c

Reply:
I missed a console error on this. Oddly, returning `null` a couple lines down fixes the SSR handling of private replays.

Reply:
Sharing the working link now to close the loop: https://devtools-git-fork-ryanjduffy-ryan-social-cards-recordreplay.vercel.app/recording/823241b9-54fd-4cc4-b315-3937694c0b0c